core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceCreator.scala

```diff
@@ -0,0 +1,688 @@
+      billingProjectOwnerPolicyEmail <- getBillingProjectOwnerPolicyEmail(workspaceRequest, span)
+      _ <- checkIfWorkspaceWithNameAlreadyExists(workspaceRequest.toWorkspaceName, span)
+      billingProject <- getBillingProjectForNewWorkspace(workspaceRequest.namespace, span)
+      (googleProjectId, googleProjectNumber) <- claimGoogleProject(billingProject, span)
+      workspace <- persistNewWorkspace(workspaceRequest, googleProjectId, googleProjectNumber, span)
+      workspaceResourceResponse <- createWorkspaceResourceInSam(workspace.workspaceId, billingProjectOwnerPolicyEmail, noWorkspaceOwner, authDomains, span)
+      policyNameToEmailMap: Map[SamResourcePolicyName, WorkbenchEmail] = workspaceResourceResponse.accessPolicies.map(x => SamResourcePolicyName(x.id.accessPolicyName) -> WorkbenchEmail(x.email)).toMap
+      _ <- createWorkflowCollectionForWorkspace(workspace.workspaceId, policyNameToEmailMap, span)
+      _ <- syncWorkspacePoliciesWithGoogle(workspace, policyNameToEmailMap, authDomains, span)
+      _ <- setupGoogleProjectForWorkspace(workspace, policyNameToEmailMap, billingProjectOwnerPolicyEmail, authDomains, span)
+      _ <- maybeUpdateGoogleProjectsInPerimeter(billingProject)
+    } yield workspace
+  }
+
+  /**
+    * Creates a new Workspace doing everything the same as createWorkspaceInternal_sequential, but this version tries
+    * to run certain steps/futures in parallel instead of running them all sequentially.  It's not super pretty.  It
+    * breaks things down into 3 different steps, each of which is its own `for` comprehension.  Why 3?  It was chosen
+    * because there was a set of steps identified that could be run first and all in parallel.  After these finish,
+    * there is a set of steps that must be executed sequentially.  After those complete, there is another set of steps
+    * that can all be kicked off in parallel.  If you need to add a new step, depending on its placement and whether it
+    * needs to be run in parallel or sequentially with existing steps, you may be able to just add it to an existing
+    * `for` comp, or you may need to add a whole new `for` comp.
+    * @param workspaceRequest
+    * @param span
+    * @return
+    */
+  private def createWorkspaceInternal_parallel(workspaceRequest: WorkspaceRequest, span: Span = null): Future[Workspace] = {
+    val authDomains = workspaceRequest.authorizationDomain.getOrElse(Set.empty)
+    val noWorkspaceOwner = workspaceRequest.noWorkspaceOwner.getOrElse(false)
+
+    for {
+      (billingProjectOwnerPolicyEmail, billingProject) <- firstSteps(workspaceRequest, span)
+      (workspace, workspaceResourceResponse) <- secondSteps(workspaceRequest, billingProject, billingProjectOwnerPolicyEmail, noWorkspaceOwner, authDomains, span)
+      _ <- thirdSteps(workspace, billingProject, workspaceResourceResponse, billingProjectOwnerPolicyEmail, authDomains, span)
+    } yield workspace
+  }
```

Comment:
thinking out loud, not for this PR: at what point is it safe to return to the user? Could we return a response to the user after `secondSteps`, then let `thirdSteps` continue asynchronously? We'd have to handle failures in the `thirdSteps` work somehow; I don't know how that  would look to the user.

Reply:
Per the other comments about ordering of things (and tests that broke) I put this back to being _mostly_ sequential.  I would love to talk about parallelizing though, but to do that, I'd rather sit with you and/or Doug and make sure we're on the same page about what can be parallelize and what needs to happen sequentially.  The good news is that _hope_ that the bulk of this refactoring has broken out the steps involved in such a way that making those decisions should hopefully be a lot easier now.